src/test/scala/ing/wbaa/druid/DruidClientSpec.scala

```diff
@@ -57,7 +59,34 @@ class DruidClientSpec extends WordSpec with Matchers with ScalaFutures {
       assertThrows[IllegalStateException] {
         config.client
       }
+    }
+
+    "throw HttpStatusException for non-200 status codes" in {
+      val config = DruidConfig(clientBackend = classOf[DruidHttpClient],
+                               hosts = Seq(QueryHost("localhost", 8086))) // yields HTTP 500
+      val client = config.client
+      val responseFuture = client.doQuery(
+        TimeSeriesQuery(
+          aggregations = List(
+            CountAggregation(name = "count")
+          ),
+          granularity = GranularityType.Hour,
+          intervals = List("2011-06-01/2017-06-01")
+        )
+      )
 
+      whenReady(responseFuture.failed) {
+        case exception: HttpStatusException =>
+          exception.status shouldBe StatusCode.int2StatusCode(500)
+          exception.entity match {
+            case Some(entity) => entity.isKnownEmpty() shouldBe true
+            case _            => fail("expected empty entity, got empty option")
+          }
+        case response => fail(s"expected HttpStatusException, got $response")
+      }
+
+      client.shutdown().futureValue
```

Comment:
In the future we should have some kind of fixture for this. But that's unrelated to this PR.